text/0007-system-collator-selection.md

```diff
@@ -0,0 +1,162 @@
+a premium. Likewise, any staked DOT for collation is _not_ staked for validation. Since collator
+sets do not need to meet Byzantine Fault Tolerance criteria, staking as the primary mechanism for
+collator selection would remove stake that is securing BFT assumptions, making the network less
+secure.
+
+Another problem with economic scalability relates to the increasing number of system chains, and
+corresponding increase in need for collators (i.e., increase in collator slots). "Good" (highly
+available, non-censoring) collators will not want to compete in elections on many chains when they
+could use their resources to compete in the more profitable validator election. Such dilution
+decreases the required bond on each chain, leaving them vulnerable to takeover by hostile
+collator groups.
+
+This RFC proposes a system whereby collation is primarily an infrastructure service, with the
+on-chain Treasury reimbursing costs of semi-trusted node operators, referred to as "Invulnerables".
+The system need not trust the individual operators, only that as a _set_ they would be resilient to
+coordinated attempts to stop a single chain from halting or to censor a particular subset of
+transactions.
+
+In the case that users do not trust this set, this RFC also proposes that each chain always have
+available collator positions that can be acquired by anyone by placing a bond.
+
+### Requirements
+
+- System MUST have at least one valid collator for every chain.
+- System MUST allow anyone to become a collator, provided they `reserve`/`hold` enough DOT.
+- System SHOULD select a set of collators with reasonable expectation that the set will not collude
+  to censor any subset of transactions.
+- Collators selected by governance SHOULD have a reasonable expectation that the Treasury will
+  reimburse their operating costs.
+
+## Stakeholders
+
+- Infrastructure providers (people who run validator/collator nodes)
+- Polkadot Treasury
+
+## Explanation
+
+This protocol builds on the existing
+[Collator Selection pallet](https://github.com/paritytech/cumulus/tree/b15da70/pallets/collator-selection)
+and its notion of Invulnerables. Invulnerables are collators (identified by their `AccountId`s) who
+will be selected as part of the collator set every session. Operations relating to the management
+of the Invulnerables are done through privileged, governance origins. The implementation should
+maintain an API for adding and removing Invulnerable collators.
+
+In addition to Invulnerables, there are also open slots for "Candidates". Anyone can register as a
+Candidate by placing a fixed bond. However, with a fixed bond and fixed number of slots, there is
+an obvious selection problem: The slots fill up without any logic to replace their occupants.
+
+This RFC proposes that the collator selection protocol allow Candidates to increase (and decrease)
+their individual bonds, sort the Candidates according to bond, and select the top `N` Candidates.
```

Comment:
Why are we not choosing randomly? Why based on the highest bond? What is the reason behind this?

Reply:
I don't really expect bonds to be high since there is not much incentive, but in case someone does have a strong incentive (e.g. thinks their transactions are being censored), they should have a path to getting selected.

With fixed bond and random selection, there are two problems (unless you see a better solution):

1. We can limit the number of candidates like we do now and select randomly. But if the set is full, how does a new candidate join? With a fixed bond, there's no comparison to justify kicking someone out.
2. We can have an unlimited number of candidates, but then people can just use multiple accounts and place the fixed bond many times, increasing their probability of being selected.

(2) could be an option, the selection rate would map to the bond amount (represented by number of accounts). It seems less efficient though as it requires running nodes for all the accounts, whereas top N requires only one node per candidate entity.

Reply:
> 2\. We can have an unlimited number of candidates, but then people can just use multiple accounts and place the fixed bond many times, increasing their probability of being selected.

Yeah that is a valid point. 

So, yeah scratch my idea.